src/ui/components/AppWrapper/AppWrapper.tsx

```diff
@@ -430,9 +426,7 @@ const AppWrapper = (props: { children: ReactNode }) => {
     // @TODO - foconnor: This is a temp hack for development to be removed pre-release.
     // These items are removed from the secure storage on re-install to re-test the on-boarding for iOS devices.
     try {
-      // @TODO - foconnor: This should use our normal DB - keeping Preferences temporarily to not break existing mobile builds.
-      // Will remove preferences again once we have better handling on APP_ALREADY_INIT with user input.
-      await PreferencesStorage.get(PreferencesKeys.APP_ALREADY_INIT);
+      await Agent.agent.basicStorage.findById(MiscRecordId.APP_ALREADY_INIT);
```

Comment:
Can you take the time to check the message in `e` and make sure its the correct one (that it's missing in the DB), and if not, throw the error

Reply:
when the key does not exist in DB it returns null, why do we use try-catch block here? and when it jumps to catch?

Reply:
Ah, with preferences storage it threw and error. In that case you should remove the try catch and do a null check instead